src/Rings/mpoly-ideals.jl

```diff
@@ -1334,24 +1334,22 @@ function minimal_generating_set(I::MPolyIdeal{<:MPolyDecRingElem}; ordering::Mon
 
   @req coefficient_ring(R) isa AbstractAlgebra.Field "The coefficient ring must be a field"
 
-  singular_assure(I, ordering)
-  IS = I.gens.S
-  RS = I.gens.Sx
-  GC.@preserve IS RS begin
-    ptr = Singular.libSingular.idMinBase(IS.ptr, RS.ptr)
-    gensS = gens(typeof(IS)(RS, ptr))
-  end
-
-  i = 1
-  while i <= length(gensS)
-    if iszero(gensS[i])
-      deleteat!(gensS, i)
-    else
-      i += 1
-    end
+  if !isempty(I.gb)
+    # make sure to not recompute a GB from scratch on the singular
+    # side if we have one
+    G = first(values(I.gb))
+    singular_assure(G, G.ord)
+    G.gens.S.isGB = true
+    _, sing_min = Singular.mstd(G.gens.S)
+    return filter(!iszero, (R).(gens(sing_min)))
+  else
+    singular_assure(I)
+    sing_gb, sing_min = Singular.mstd(I.gens.gens.S)
+    ring = I.gens.Ox
+    computed_gb = IdealGens(ring, sing_gb, true)
+    I.gb[eval(ordering(ring))(ring)] = computed_gb
```

Comment:
Wait, `eval`? That sounds super fragile; also can't orderings be more than symbols these days?

Also, what is the difference between `ordering(ring)` and `default_ordering(ring`) ?

Reply:
I removed the call to `eval` just know, yesterday doing it like this felt good but I can't remember why (and safe because `ordering` always returns a symbol).